addons/l10n_my_edi/models/account_edi_xml_ubl_my.py

```diff
@@ -159,6 +170,31 @@ def _export_invoice_vals(self, invoice):
 
         return vals
 
+    def _decode_myinvois_attachment(self, attachment):
+        """ Extract data from MyInvois xml. """
+        def get_node(node, xpath):
+            nodes = node.xpath(xpath)
+            return nodes[0] if nodes else None
+
+        def get_value(node):
+            if node is None:
+                return None
+            return node.text
+
+        vals = {}
+        try:
+            root = etree.fromstring(attachment.raw)
+            invoice_id_node = get_node(root, "//*[local-name()='Invoice']/*[local-name()='ID']")
+        except etree.XMLSyntaxError:
+            # Not an xml
+            return {}
+        except AttributeError:
+            # Not a MyInvois xml
+            return {}
+
+        vals['original_document_id'] = get_value(invoice_id_node)
+        return vals
```

Comment:
Why return a dict of a single value here instead of directly returning the id?

Reply:
Maybe it won't be the only value we'll need to recover from the XML, so I left it more modular for future use.
Feel free to disagree :D